### PR TITLE
home page analytics - track home screen buttons

### PR DESCRIPTION
**Pull request by jyothishnt**
## Type

- [ ] Bug fix
- [x] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1409

## Description
Track home screen buttons

## Deployment URL
http://homepage-analytics.review.ensembl.org/

## Views affected
2020 Home screen landing page

### Other effects
NA

- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
NA

### CODE REVIEW

**Comment by ens-ma7** [comment]
i have tried the chrome extension for google analytics and see is what i am seeing:
`analytics.js:26 Executing Google Analytics commands.
analytics.js:26 Running command: ga("send", {hitType: "event", eventCategory: "homepage-action-button", eventAction: "clicked", eventLabel: "entity-viewer", transport: "xhr"})
instrument.js:109 Setting throttling cookie: "_gat"
analytics.js:26 Executing Google Analytics commands.
analytics.js:26 Running command: ga("send", {hitType: "pageview", page: "/entity-viewer"})
analytics.js:26 Executing Google Analytics commands.
analytics.js:26 Running command: ga("set", "dimension2", "")
analytics.js:26 Executing Google Analytics commands.
analytics.js:26 Running command: ga("set", "dimension2", "entity-viewer")`
It seems to be working fine but my only concern is where it is setting a blank value for dimension2:
`analytics.js:26 Running command: ga("set", "dimension2", "")`

Not sure if this is whats happening, if thats the case, i wonder if there will be an obsolete blank value in analytics.

**Review by ens-ma7 — APPROVED** [review]

**Review by imransl — APPROVED** [review]

File: src/ensembl/src/content/home/hooks/useHomeAnalytics.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const trackButtonsOnScreen = (buttonText: string) => {
+    analyticsTracking.trackEvent({
+      category: 'homepage-action-button',
+      action: 'clicked',
+      label: buttonText
+    });
+  };
+
+  return {
+    trackButtonsOnScreen
```
Comment (azangru) [inline src/ensembl/src/content/home/hooks/useHomeAnalytics.ts:29]: This is a strange name 😕 Presumably, every button is on the screen; where else can they be?

`trackNavigationToApp`? `trackCTAButtonClick`?

Comment (jyothishnt) [reply src/ensembl/src/content/home/hooks/useHomeAnalytics.ts:29]: What is CTA? :-)
OK, I will go with `trackHomePageActionButtons` as per the ga category?

Comment (azangru) [reply src/ensembl/src/content/home/hooks/useHomeAnalytics.ts:29]: > What is CTA?

Call to action

**Review by azangru — APPROVED** [review]

**Pull request merged**